### DISCUSSION TITLE
yarn2 Vercel build failed - No Next.js version could be detected in your project

### DISCUSSION THREAD

**Discussion by wenerme** [vercel/vercel#4910] [Help]
log here https://vercel.com/wener/yarn-berry-next-starter/9lpgz5zfz

I think yarn2 is supported now https://github.com/vercel/vercel/issues/4436#issuecomment-662737381

Use latest for next dependency https://github.com/wenerme/node-yarn-starter/blob/master/packages/server/package.json

**Comment by rodrigojcmello** [comment]
some here

**Reply by shouze** [reply]
same here, looks like because the builder is trying to guess next version by looking into node_modules directory:
https://github.com/vercel/vercel/blob/c7e2280d231443917ecab2fffefdf0aea2a5bb78/packages/now-next/src/index.ts#L144-L154

maybe it worth to try some [`require.resolve()`](https://yarnpkg.com/features/pnp#packages-are-stored-inside-zip-archives-how-can-i-access-their-files) for yarn2 compat? :thinking: 
![image](https://user-images.githubusercontent.com/54712/91823709-ebb4b100-ec39-11ea-9615-2a6125fd28e5.png)

**Reply by fmal** [reply]
I'm running into same issue with a monorepo setup using yarn berry (with offline cache and pnp) and having next.js added as a dependency in a workspace (not root) 😠

**Reply by kachkaev** [reply]
@shouze in theory, this could be fixed by a new version of [`resolve-from`](https://www.npmjs.com/package/resolve-from), but the author does not seem to be keen to support Yarn 2: https://github.com/sindresorhus/resolve-from/issues/16

**Reply by shouze** [reply]
@kachkaev good catch!
I understand but given how and why `resolveFrom` is used to (retrieve next.js version from its package.json file)... I guess that we could try to decorate `resolveFrom` by following yarn2 recommandations no?

**Reply by shouze** [reply]
Or even better.... current package could be replaced by a pnp compatible one like https://github.com/browserify/resolve.

Yarn provides a [compatibility table](https://yarnpkg.com/features/pnp#native-support).

**Reply by shouze** [reply]
@Timer as you're the last editor of this part of file in https://github.com/vercel/vercel/pull/3431 WDYT?

**Reply by shouze** [reply]
And also @flybayer as you also contributed through https://github.com/vercel/vercel/pull/4465 WDYT?

**Reply by kachkaev** [reply]
Let‘s also see if we can get some help from the Yarn 2 community: https://github.com/yarnpkg/berry/issues/2031 🤞
